tools/utils/BPE.lua

```diff
@@ -14,8 +13,25 @@ function BPE:__init(codesfile_path, joiner_annotate, joiner_new)
     end
   end
   self.codes = {}
-  local f = assert(io.open(codesfile_path, "r"))
+  local f = assert(io.open(opt.bpe_model, "r"))
+
+  self.EOT_marker = opt.EOT_marker
+  self.BOT_marker = opt.BOT_marker
+  self.joiner_new = opt.joiner_new
+  self.joiner_annotate = opt.joiner_annotate
+
   local t = f:read("*line")
+  local options = self.split(t, ";")
+  if (#options == 4) then
+    self.prefix = options[1] == "true" and true or false
```

Comment:
thanks Yongchao - I did not ask where it was coming from, but why you were using this tautological expression...